stingray/gti.py

```diff
@@ -1152,51 +1266,210 @@ def gti_border_bins(gtis, time, dt=None, epsilon=0.001):
     --------
     >>> times = np.arange(0.5, 13.5)
 
+    >>> gti_border_bins([[16., 18.]], times)
+    Traceback (most recent call last):
+        ...
+    ValueError: Invalid time interval for the given GTIs
+
     >>> start_bins, stop_bins = gti_border_bins(
     ...    [[0, 5], [6, 8]], times)
 
     >>> np.allclose(start_bins, [0, 6])
     True
     >>> np.allclose(stop_bins, [5, 8])
     True
-    >>> np.allclose(times[start_bins[0]:stop_bins[0]], [ 0.5, 1.5, 2.5, 3.5, 4.5])
+    >>> np.allclose(times[start_bins[0]:stop_bins[0]], [0.5, 1.5, 2.5, 3.5, 4.5])
     True
     >>> np.allclose(times[start_bins[1]:stop_bins[1]], [6.5, 7.5])
     True
     """
+    time = np.asarray(time)
+    gtis = np.asarray(gtis)
     if dt is None:
         dt = np.median(np.diff(time))
 
     epsilon_times_dt = epsilon * dt
 
+    if time[-1] < np.min(gtis) or time[0] > np.max(gtis):
+        raise ValueError("Invalid time interval for the given GTIs")
+
     spectrum_start_bins = []
     spectrum_stop_bins = []
 
-    for g in gtis:
-        # Epsilons here resolve a numerical problem when time edges end up
-        # outside GTIs.
-        good = ((time - dt / 2 >= g[0] - epsilon_times_dt) &
-                (time + dt / 2 <= g[1] + epsilon_times_dt))
-        t_good = time[good]
-        if len(t_good) == 0:
-            continue
-        startbin = np.argmin(np.abs(time - dt / 2 - g[0]))
-        stopbin = np.searchsorted(time + dt / 2, g[1], 'right') + 1
-        if stopbin > len(time):
-            stopbin = len(time)
+    gti_low = gtis[:, 0] + dt / 2 - epsilon_times_dt
```

Comment:
This whole section of code below seems to be a duplication of that in the function above. Can we pull this out into its own function for use in both rather than duplicate the code?

Reply:
There are slight differences, that - I think- would make the code less readable if made into a single function with `if`s for the various cases